account_financial_report_webkit/models/account_static_balance.py

```diff
@@ -53,119 +61,144 @@ def unlink(self):
         """ When manually unlinking, trigger a recalculation of the entry. """
         accounts = self.mapped('account_id')
         periods = self.mapped('period_id')
+        journals = self.mapped('journal_id')
         res = super(AccountStaticBalance, self).unlink()
         if not self.env.context.get('skip_recalculation', False):
             # Trigger recalculation of the entry
             self.env['account.static.balance'].calculate_static_balance(
-                periods=periods, accounts=accounts)
+                periods=periods, accounts=accounts, journals=journals)
         return res
 
     @api.model
     def auto_populate_table(self):
         """ ir.cron function to trigger intial balance calculation for all
         closed periods and accounts if their combination is missing. """
-        periods, accounts = self.get_missing_periods_and_accounts()
+        periods, accounts, journals = \
+            self.get_missing_periods_accounts_and_journals()
         if not periods:
             logger.debug("No missing entries found. Static account balance "
                          "data is up to date.")
             return
-        self.calculate_static_balance(periods=periods, accounts=accounts)
+        self.calculate_static_balance(
+            periods=periods, accounts=accounts, journals=journals)
 
     @api.multi
-    def get_missing_periods_and_accounts(self, periods=None):
+    def get_missing_periods_accounts_and_journals(self, periods=None):
         # Split into unique periods and accounts
-        per_ids = acc_ids = []
+        per_ids = []
+        acc_ids = []
+        journal_ids = []
         missing_combinations = self.get_entries_to_calculate(periods=periods)
         if missing_combinations:
-            per_ids, acc_ids = map(list, map(set, zip(*missing_combinations)))
+            per_ids, acc_ids, journal_ids = map(
+                list, map(set, zip(*missing_combinations)))
```

Comment:
I think the zipping is the problem here in the original design, and it's become more of a problem now that we are adding journals into the mix. `get_entries_to_calculate` gives us the journals per period that are missing, but we flatten that information. Do you think it would be better if we used that information and only regenerated the missing combinations? That might only make sense if the number of missing journals for a given period is smallish. There might be a cutoff point beyond which it is no longer beneficial, if at all.